utilities/include/alps/numeric/tensors/tensor_base.hpp

```diff
@@ -207,17 +207,70 @@ namespace alps {
         template<typename T2, typename St, typename = std::enable_if<std::is_same<Container, storageType>::value, void >>
         tensor_base(tensor_base<T2, Dim, St> &&rhs) noexcept: storage_(rhs.storage()), shape_(rhs.shape()), acc_sizes_(rhs.acc_sizes()) {}
 
-        /// Different type assignment
-        template<typename T2, typename St>
-        tensor_base < T, Dim, Container > &operator=(const tensor_base < T2, Dim, St> &rhs){
-          assert(size()==rhs.size());
+        // Different type assignments
+
+        // assign when lhs and rhs containers are same
+        /// Copy assignment
+        tensor_base < T, Dim, Container > & operator=(const tensor_base < T, Dim, Container > &rhs) {
+          shape_ = rhs.shape();
+          acc_sizes_ = rhs.acc_sizes();
           storage_ = rhs.storage();
           return *this;
-        };
+        }
+        /// Move assignment
+        tensor_base < T, Dim, Container > & operator=(tensor_base < T, Dim, Container > &&rhs) {
+          shape_ = std::move(rhs.shape());
+          acc_sizes_ = std::move(rhs.acc_sizes());
+          storage_ = std::move(rhs.storage());
```

Comment:
Question: What happens if `storage_` is a view?

Reply:
`data_view` stores data in the `view` object, and it has deep copy for move assignment, because we can not move pointers between views.